### PR TITLE
chore(logging): refactor plugins to use logger only from provider COMPASS-7407 COMPASS-7315

### PR DESCRIPTION
**Pull request by gribnoysup**
This patch removes direct usage of compass-logging service in plugin stores. To make sure we don't introduce any regressions in the future I also added a special plugin eslint config that prohibits non-provider imports from the service packages (not all of them as we are still working on it)

In most cases there is no changes in current compass behavior, but in some cases where propagating `debug` usage was a bit too much of a hassle I just removed it, if we feel like those have any value, we should probably replace them with `log.debug` calls instead, as debug just ends up in the devtools console where I don't think many of us are looking at it. Happy to do that change if you feel like it's worth it.

### CODE REVIEW

File: packages/compass-logging/src/provider.ts [hunk 1]
```diff
@@ -110,6 +110,8 @@ export function withLoggerAndTelemetry<
   return WithLoggerAndTelemetry;
 }
 
-export function mongoLogId(id: number): MongoLogId {
+// Because dupedLogId needs to stay on the same line
+// prettier-ignore
+export function mongoLogId(id: number): MongoLogId { // !dupedLogId
```
Comment (gribnoysup) [inline packages/compass-logging/src/provider.ts:115]: Not technically duped, but this is what we use to skip check-logids check in mongosh and eslint rule and I don't want to add another way of skipping it

File: packages/compass-crud/src/stores/crud-store.ts [hunk 2]
```diff
@@ -1057,7 +1064,7 @@ class CrudStoreImpl
   }
 
   /**
-   * Open the insert document dialog.
+   * Open the insert document diathis.logger.log.
```
Comment (gribnoysup) [inline packages/compass-crud/src/stores/crud-store.ts:1067]:
```suggestion
   * Open the insert document dialog.
```

**Review by addaleax — APPROVED** [review]
Looks good so far!

**Comment by gribnoysup** [comment]
So CI is passing mostly, but macos machines are a bottleneck now, so I'll merge without waiting for it to go fully green

**Pull request merged**